cli/src/commands/backend/deploy.rs

```diff
@@ -0,0 +1,372 @@
+					.insert(env.environment_id, env_config.clone());
+			})
+			.await?;
+		}
+
+		rivet_term::status::info("Migrating databases", "");
+
+		let mut migrate_cmd = Command::new("opengb");
+		migrate_cmd.current_dir(&path);
+		migrate_cmd.arg("db").arg("deploy");
+
+		// Insert all database urls into env
+		for (db_name, db) in &env_config.databases {
+			migrate_cmd.env(
+				format!("DATABASE_URL_{}", db_name),
+				db.url.clone(),
+			);
+		}
+
+		ensure!(
+			migrate_cmd.status().await?.success(),
+			"Failed to migrate OpenGB databases"
+		);
+
+		// Read files for upload
+		let gen_manifest = read_gen_manifest(&path).await?;
+		let files = vec![
+			upload::prepare_upload_file(
+				&gen_manifest.bundle,
+				"bundle.js",
+				fs::metadata(&gen_manifest.bundle).await?,
+			)?,
+			// TODO: Get rid of unwrap
+			upload::prepare_upload_file(
+				gen_manifest.wasm.as_ref().unwrap(),
+				"query-engine.wasm",
+				fs::metadata(gen_manifest.wasm.as_ref().unwrap()).await?,
+			)?,
+		];
+		let total_len = files
+			.iter()
+			.fold(0, |acc, x| acc + x.prepared.content_length);
+
+		eprintln!();
+		rivet_term::status::info(
+			"Uploading Environment",
+			format!(
+				"{name} ({count} files, {size} total)",
+				name = &env.display_name,
+				count = files.len(),
+				size = upload::format_file_size(total_len as u64)?,
+			),
+		);
+
+		let prepare_res = unwrap!(
+			apis::ee_cloud_opengb_projects_envs_api::ee_cloud_opengb_projects_envs_prepare_deploy(
+				&ctx.openapi_config_cloud,
+				&project_id,
+				&env_id,
+				models::EeCloudOpengbProjectsEnvsPrepareDeployRequest {
+					files: files.iter().map(|f| f.prepared.clone()).collect(),
+				},
+			)
+			.await
+		);
+
+		// Upload files
+		let reqwest_client = Arc::new(reqwest::Client::new());
+		let pb = term::EitherProgressBar::Multi(indicatif::MultiProgress::new());
+
+		futures_util::stream::iter(prepare_res.presigned_requests)
+			.map(Ok)
+			.try_for_each_concurrent(8, |presigned_req| {
+				let pb = pb.clone();
+				let files = files.clone();
+				let reqwest_client = reqwest_client.clone();
+
+				async move {
+					// Find the matching prepared file
+					let file = unwrap!(
+						files.iter().find(|f| f.prepared.path == presigned_req.path),
+						"missing prepared file"
+					);
+
+					upload::upload_file(
+						&reqwest_client,
+						&presigned_req,
+						&file.absolute_path,
+						file.prepared.content_type.as_ref(),
+						pb,
+					)
+					.await?;
+
+					GlobalResult::<()>::Ok(())
+				}
+			})
+			.await?;
+
+		eprintln!("\n");
+		rivet_term::status::info("Deploying environment", &env.display_name);
+
+		unwrap!(
+			apis::ee_cloud_opengb_projects_envs_api::ee_cloud_opengb_projects_envs_deploy(
+				&ctx.openapi_config_cloud,
+				&project_id,
+				&env_id,
+				models::EeCloudOpengbProjectsEnvsDeployRequest {
+					modules,
+					upload_id: prepare_res.upload_id,
+				},
+			)
+			.await
+		);
+
+		let api = format!(
+			"https://{}--{}.opengb.{}/",
```

Comment:
is there a clean way to pull the script name from the server instead of doing this? or better have the server pass the full url.